builtin/providers/azurerm/resource_arm_eventhub_namespace.go

```diff
@@ -169,9 +170,15 @@ func resourceArmEventHubNamespaceDelete(d *schema.ResourceData, meta interface{}
 	resGroup := id.ResourceGroup
 	name := id.Path["namespaces"]
 
-	resp, err := namespaceClient.Delete(resGroup, name, make(chan struct{}))
+	delResp, error := namespaceClient.Delete(resGroup, name, make(chan struct{}))
+	resp := <-delResp
+	err = <-error
 
-	if resp.StatusCode != http.StatusNotFound {
+	if resp.StatusCode == http.StatusNotFound {
```

Comment:
Looking into this, internally the Azure SDK is polling for one of the status codes:

```
azure.WithErrorUnlessStatusCode(http.StatusNoContent, http.StatusOK, http.StatusAccepted)
```

As such, I've removed this if statement - as we can lean on the SDK to check for the Status Code errors here (which are returned as an error and will be handled below)